### PR TITLE
Install and run some Go linters

### PR DESCRIPTION
**Pull request by zimmski**

### CODE REVIEW

File: evaluate/metrics/testing/assessments.go [hunk 1]
```diff
@@ -11,40 +11,40 @@ import (
 
 // Clean deletes all empty and nondeterministic keys from the assessment.
 func Clean(assessment metrics.Assessments) metrics.Assessments {
-	copy := metrics.Assessments{}
-	maps.Copy(copy, assessment)
+	c := metrics.Assessments{}
```
Comment (bauersimon) [inline evaluate/metrics/testing/assessments.go:14]: unnecessary change but not a big deal

File: evaluate/report/csv.go [hunk 2]
```diff
@@ -84,12 +84,18 @@ func RecordsFromEvaluationCSVFiles(evaluationCSVFilePaths []string) (records [][
 		if err != nil {
 			return nil, pkgerrors.WithStack(err)
 		}
-		defer file.Close()
+		defer func() {
+			if err := file.Close(); err != nil {
+				panic(err)
```
Comment (bauersimon) [inline evaluate/report/csv.go:89]: plz return error as return value

File: evaluate/report/csv.go [hunk 3]
```diff
@@ -84,12 +84,18 @@ func RecordsFromEvaluationCSVFiles(evaluationCSVFilePaths []string) (records [][
...
+			}
+		}()
 
 		csv := csv.NewReader(file)
 
 		// Ignore the CSV header.
-		csv.Read()
+		if _, err := csv.Read(); err != nil {
+			panic(err)
```
Comment (bauersimon) [inline evaluate/report/csv.go:97]: plz return error as return value

File: log/logger.go [hunk 4]
```diff
@@ -306,13 +319,15 @@ func (h *spawningHandler) WithAttrs(attributes []slog.Attr) slog.Handler {
...
-			fmt.Fprintf(h.writer, "ERROR: cannot create new handler: %s\n", err.Error())
+			_, _ = fmt.Fprintf(h.writer, "ERROR: cannot create new handler: %s\n", err.Error())
 
 			continue
 		}
 
 		logMessage := fmt.Sprintf("Spawning new log file at %s", logFilePath)
-		h.Handle(context.Background(), slog.NewRecord(time.Now(), slog.LevelInfo, logMessage, 0))
+		if err := h.Handle(context.Background(), slog.NewRecord(time.Now(), slog.LevelInfo, logMessage, 0)); err != nil {
+			panic(err)
```
Comment (bauersimon) [inline log/logger.go:329]: plz adapt to the same error handling from above:

```go
_, _ = fmt.Fprintf(h.writer, "ERROR: cannot create new handler: %s\n", err.Error())

continue
```

File: provider/openrouter/openrouter.go [hunk 5]
```diff
@@ -96,7 +96,11 @@ func (p *Provider) fetchModels() (models ModelsList, err error) {
 			if err != nil {
 				return pkgerrors.WithStack(err)
 			}
-			defer response.Body.Close()
+			defer func() {
+				if err := response.Body.Close(); err != nil {
+					panic(err)
```
Comment (bauersimon) [inline provider/openrouter/openrouter.go:101]: plz return error as return value

File: scripts/ollama-models/main.go [hunk 6]
```diff
@@ -100,32 +104,45 @@ func main() {
 	if err != nil {
 		log.Fatalf("failed creating file: %s", err)
 	}
-	defer file.Close()
+	defer func() {
+		if err := file.Close(); err != nil {
+			panic(err)
```
Comment (bauersimon) [inline scripts/ollama-models/main.go:109]: this script uses `log.Fatalf` for all errors, not too important though

**Review by bauersimon — CHANGES_REQUESTED** [review]

**Comment by bauersimon** [comment]
image

File: evaluate/metrics/testing/assessments.go (hunk 1 above)
Comment (zimmski) [reply evaluate/metrics/testing/assessments.go:14]: Necessary change: do not overwrite keywords/Go functions: "copy"

**Review by bauersimon — APPROVED** [review]
Wonderful :fr:
